File: python/series/series.py
def slices(series, length):
  if length == 0:
    raise ValueError('Test slice cannot be zero')
  if length < 0:
    raise ValueError('Test slice cannot be negative')
  if len(series) == 0:
    raise ValueError('Empty series is invalid')
  if length > len(series):
    raise ValueError('Series does not have enough characters')

  series_list = list(series)
  return [''.join(series_list[i:i+length]) for i in range(len(series_list) - length + 1)]

File: csharp/series/Series.cs
using System;
using System.Text;

public static class Series
{
    public static string[] Slices(string numbers, int sliceLength)
    {
        if (sliceLength > numbers.Length || sliceLength <= 0) {
            throw new ArgumentException("Slice length is invalid.");
        } else if (numbers.Length == 0) {
            throw new ArgumentException("Empty seties is invalid.");
        }
        string[] slices = new string[numbers.Length - sliceLength + 1];
        for (int i = 0; i < numbers.Length - sliceLength + 1; i++) {
            StringBuilder slice = new StringBuilder();
            for (int j = 0; j < sliceLength; j++) {
                slice.Append(numbers[i+j]);
            }
            slices[i] = slice.ToString();
        }
        return slices;
    }
}